### PR TITLE
Retain scheme in LBClient.RibbonRequest URI - fixes #183

### LINKED ISSUE #183: RibbonClient doesn't support HTTPS
Using the following test - https://gist.github.com/bstick12/2a21ffd30b14ec68716f#file-ribbonclienttest

You will see the following exception raised in the logs

```
21:04:25.583 [pool-1-thread-2] DEBUG c.n.l.LoadBalancerExecutor - Got error java.net.SocketException: Unexpected end of file from server when executed on server api.github.com:443
```
This is due to the request being made to http://api.github.com:443/. 

The URI created in RibbonClient does include a scheme and eventually com.netflix.loadbalancer.LoadBalancerContext.deriveSchemeAndPortFromPartialUri(URI) will default the scheme to http.

### PR DESCRIPTION
**Pull request by bstick12**

### CODE REVIEW

**Comment by adriancole** [comment]
Thanks for the help. I think we need a test for this, right? Can you port something like this into the ribbon test? That way future work doesn't undo yours. https://github.com/Netflix/feign/blob/master/core/src/test/java/feign/client/DefaultClientTest.java#L119

**Comment by bstick12** [comment]
Added a test to check the https connection works

File: ribbon/src/test/java/feign/ribbon/RibbonClientTest.java [hunk 1]
```diff
@@ -123,6 +125,26 @@ public void urlEncodeQueryStringParameters() throws IOException, InterruptedExce
     assertEquals(recordedRequestLine, expectedRequestLine);
   }
 
+
+  @Test
+  // Issue-183
```
Comment (adriancole) [inline ribbon/src/test/java/feign/ribbon/RibbonClientTest.java]: nit I don't think this comment is required. the commit log already says this, and the need for this feature is obvious.

**Comment by adriancole** [comment]
hypernit on comment, also I'm trying to help us get more readable commit log. can you switch to this form?
Note using retains vs retain makes the commit log look like.

"this commit retains scheme in LBClient.RibbonRequest"

```
Retains scheme in LBClient.RibbonRequest URI

Before this change, we were dropping scheme, which prevented use of https.

closes #183
``

**Pull request merged**